src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,221 @@
+import duke.command.ListCommand;
+import duke.command.add.DeadlineCommand;
+import duke.command.add.EventCommand;
+import duke.command.add.TodoCommand;
+import duke.command.index.DeleteCommand;
+import duke.command.index.MarkCommand;
+import duke.exception.DukeException;
+import duke.exception.InvalidArgumentException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Represents a system to translate user input into respective objects.
+ * <code>Parser</code> class methods are static, cannot be created and
+ * is used to convert String of user input into its intended object.
+ */
+public class Parser {
+    public static final String DATE_INPUT_FORMAT = "dd-MM-yyyy";
+    public static final String TIME_INPUT_FORMAT = "HHmm";
+
+    /**
+     * Returns a LocalDate object specified by the user input.
+     * @param input date input.
+     * @return a LocalDate object of the specified date.
+     */
+    public static LocalDate convertToDate(String input) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_INPUT_FORMAT);
+
+        return LocalDate.parse(input, formatter);
+    }
+
+    /**
+     * Returns a LocalTime object specified by the user input.
+     * @param input time input.
+     * @return a LocalTime object of the specified time.
+     */
+    public static LocalTime convertToTime(String input) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(TIME_INPUT_FORMAT);
+
+        return LocalTime.parse(input, formatter);
+    }
+
+    /**
+     * Returns a Command object based on the user input.
+     *
+     * @param input user command input.
+     * @return a command object.
+     * @throws DukeException if command format is invalid.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        String[] inputArgs = input.trim().split(" ", 2);
+
+        switch (inputArgs[0]) {
+        case TodoCommand.COMMAND_WORD:
+            return handleTodo(inputArgs);
+        case DeadlineCommand.COMMAND_WORD:
+            return handleDeadline(inputArgs);
+        case EventCommand.COMMAND_WORD:
+            return handleEvent(inputArgs);
+        case DeleteCommand.COMMAND_WORD:
+            return handleDelete(inputArgs);
+        case MarkCommand.COMMAND_WORD_MARK:
+            return handleMark(inputArgs, true);
+        case MarkCommand.COMMAND_WORD_UNMARK:
+            return handleMark(inputArgs, false);
+        case ListCommand.COMMAND_WORD:
+            return handleList(inputArgs);
+        case FindCommand.COMMAND_WORD:
+            return handleFind(inputArgs);
+        case ExitCommand.COMMAND_WORD:
+            return handleExit(inputArgs);
+        case BotCommand.COMMAND_WORD_JJBA:
+            return handleBot(inputArgs, BotType.JJBA);
+        case BotCommand.COMMAND_WORD_DIO:
+            return handleBot(inputArgs, BotType.DIO);
+        default:
+            throw new InvalidArgumentException();
+        }
```

Comment:
I like how this is abstracted into a `Parser` class, this simplifies the rest of the other classes when parsing a duke command. 👍